cpp/inc/bond/core/exception.h

```diff
@@ -85,6 +119,13 @@ BOND_NORETURN inline void NothingException()
 }
 
 
+BOND_NORETURN inline void ExcessiveAllocationException()
```

Comment:
This isn't being used at the moment. What are the pros/cons for adding it now instead of waiting until we use it in some other code?